src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -684,36 +653,66 @@ private static ISymbol GetEventAccessor(IEventSymbol iEvent, IOperation operatio
+            if (symbol.ContainingSymbol is INamedTypeSymbol namedType && namedType.IsGenericType)
             {
-                if (TryGetOrCreatePlatformAttributes(symbol, platformSpecificMembers, out var operationAttributes))
+                CheckTypeArguments(namedType.TypeArguments, operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, PooledHashSet<ITypeSymbol>.GetInstance());
+            }
+
+            static void CheckTypeArguments(ImmutableArray<ITypeSymbol> typeArguments, IOperation operation, OperationAnalysisContext context,
+                PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>, SmallDictionary<string, PlatformAttributes>> platformSpecificOperations,
+                ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers, ImmutableArray<string> msBuildPlatforms,
+                PooledHashSet<ITypeSymbol> visitedTypes)
+            {
+                foreach (var typeArgument in typeArguments)
+                {
+                    if (!visitedTypes.Contains(typeArgument))
+                    {
+                        visitedTypes.Add(typeArgument);
+                        if (typeArgument.SpecialType == SpecialType.None)
+                        {
+                            CheckOperationAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, typeArgument, checkParents: true);
+
+                            if (typeArgument is INamedTypeSymbol nType && nType.IsGenericType)
+                            {
+                                CheckTypeArguments(nType.TypeArguments, operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, visitedTypes);
+                            }
+                        }
+                    }
+                }
+            }
+
+            static void CheckOperationAttributes(IOperation operation, OperationAnalysisContext context, PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>,
+                SmallDictionary<string, PlatformAttributes>> platformSpecificOperations, ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers,
+                ImmutableArray<string> msBuildPlatforms, ISymbol symbol, bool checkParents)
+            {
+                if (TryGetOrCreatePlatformAttributes(symbol, checkParents, platformSpecificMembers, out var operationAttributes))
                 {
-                    if (TryGetOrCreatePlatformAttributes(context.ContainingSymbol, platformSpecificMembers, out var callSiteAttributes))
+                    var containingSymbol = context.ContainingSymbol;
+                    if (containingSymbol is IMethodSymbol method && method.IsAccessorMethod())
+                    {
+                        containingSymbol = method.AssociatedSymbol;
```

Comment:
As we are not adding parent attributes to the accessor methods so that it will not report a duplicate warning when the containing symbol had any attribute (for example property has no accessor level attribute but the entire property has attribute). But when the property getter or setter any statements inside it became the containing symbol for those statements. When accessors used as call site, as their attributes are not including the parent attributes call site attributes are incomplete and causing warnings. So here when accessor is the containing symbol change it to the AssociatedSymbol which is the real Property or Event which has complete attributes for call site